plugins/nf-tower/src/main/io/seqera/tower/plugin/TowerFusionEnv.groovy

```diff
@@ -0,0 +1,266 @@
+            .header('Content-Type', 'application/json')
+            .header('Authorization', "Bearer ${accessToken}")
+            .POST(
+                HttpRequest.BodyPublishers.ofString(
+                    new Gson().toJson(
+                        new LicenseTokenRequest(
+                            product: product,
+                            version: version
+                        ),
+                        LicenseTokenRequest.class
+                    ),
+                )
+            )
+            .build()
+
+        try {
+            final resp = safeHttpSend(req, retryPolicy)
+
+            if (resp.statusCode() == 200) {
+                return new Gson().fromJson(resp.body(), LicenseTokenResponse.class).signedToken
+            }
+
+            if (resp.statusCode() == 401) {
+                throw new UnauthorizedException("Unauthorized [401] - Verify you have provided a valid access token")
+            }
+
+            throw new BadResponseException("Invalid response: ${req.method()} ${req.uri()} [${resp.statusCode()}] ${resp.body()}")
+
+        } catch (IOException e) {
+            throw new IllegalStateException("Unable to send request to '${req.uri()}' : ${e.message}")
+        }
+    }
+
+    /**************************************************************************
+     * Helper methods
+     *************************************************************************/
+
+    /**
+     * Get the configured Platform API endpoint: if the endpoint is not provided in the configuration, we fallback to the
+     * environment variable `TOWER_API_ENDPOINT`. If neither is provided, we fallback to the default endpoint.
+     *
+     * @param opts the configuration options for Platform
+     * @param env the applicable environment variables
+     * @return the Platform API endpoint
+     */
+    protected static String endpoint0(Map opts, Map<String, String> env) {
+        def result = opts.endpoint as String
+        if (!result || result == '-') {
+            result = env.get('TOWER_API_ENDPOINT') ?: TowerClient.DEF_ENDPOINT_URL
+        }
+        return result.stripEnd('/')
+    }
+
+    /**
+     * Get the configured Platform access token: if `TOWER_WORKFLOW_ID` is provided in the environment, we are running
+     * in a Platform-made run and we should ONLY retrieve the token from the environment. Otherwise, check
+     * the configuration file or fallback to the environment. If no token is found, returns null.
+     *
+     * @param opts the configuration options for Platform
+     * @param env the applicable environment variables
+     * @return the Platform access token
+     */
+    protected static String accessToken0(Map opts, Map<String, String> env) {
+        def token = env.get('TOWER_WORKFLOW_ID')
+            ? env.get('TOWER_ACCESS_TOKEN')
+            : opts.containsKey('accessToken') ? opts.accessToken as String : env.get('TOWER_ACCESS_TOKEN')
+        return token
+    }
```

Comment:
This basically replicates what `TowerConfig` does in `nf-wave`, but I didn't want to add a explicit dependency on that.

Reply:
Could make sense to turn those into `PlatformHelper` class shared across modules & plugins

Reply:
I agree. Do you want it done in the context of this PR?

Reply:
it could be better

Reply:
Refactored with c1147cdc922a9a5423e1ea5131cbf7c5a70ce497. I created the `PlatformHelper` class under `modules/nextflow/src/main/groovy/nextflow/platform` since it seemed to fit with the existing conventions for common code. Please double check.